adapter/src/dbt/adapters/fal_experimental/support/trino.py

```diff
@@ -0,0 +1,34 @@
+from dbt.adapters.base import BaseAdapter, BaseRelation
+from dbt.adapters.base.connections import AdapterResponse
+from dbt.adapters.fal_experimental.adapter_support import new_connection
+import pandas as pd
+from dbt.adapters.sql import SQLAdapter
+from trino.dbapi import connect
+
+
+def read_relation_as_df(adapter: BaseAdapter, relation: BaseRelation) -> pd.DataFrame:
+    creds = adapter.config.credentials._db_creds
+    with connect(
+        host=creds.host,
+        port=creds.port,
+        user=creds.user,
+        catalog=creds.database,
+        schema=creds.schema,
+    ) as conn:
```

Comment:
Ideally we don't build a connection in a manual way in case the `_db_creds` change. Can we use the `new_connection` function in adapter_support.py?